api/helpers/game-states/validate-all-cards.js

```diff
@@ -0,0 +1,64 @@
+const DeckIds = require('../../../utils/DeckIds.json');
+
+module.exports = {
+  friendlyName: 'Validate All Cards',
+
+  description:
+    'Validate that all the cards are accounted for and there are no duplicates, and that only face and pointcards can have attachments',
+
+  inputs: {
+    gameState: {
+      type: 'ref',
+      description: 'Current GameState',
+      required: true,
+    },
+  },
+  sync: true,
+
+  fn: ({ gameState }, exits) => {
+    const { p0, p1, deck, scrap, twos, oneOff } = gameState;
+
+    // put Cards and Attachments into a one dimensional array
```

Comment:
The idea here was to only flatten out the ones that were allowed to have `attachments`, this way if any other show that they do have attachments, we know they shouldn't.